### ISSUE TITLE
Consider using an AONT (maybe even instead of snappy?)

### ISSUE THREAD

**Issue by eternaleye** [LogoiLab/PaddedCell#4]
An AONT, or All-Or-Nothing Transform, makes it impossible to recover content if any bit of it is missing or changed. Common implementations use encryption in a way that _also_ renders the content unrecognizable, which may meet your goal of rendering it "safe".

A common approach is:

1. Generate a random key _K_
2. Encrypt the plaintext using _K_ to obtain a ciphertext _C_
3. Hash _C_ to obtain _H_
4. XOR _K_ with _H_ to obtain a tag _T_
5. Output _C_ concatenated with _T_

Unpacking is simply:

1. Split the data into _C_ and _T_ (_T_ is a fixed length)
2. Hash _C_ to obtain _H_
3. XOR _T_ with _H_ to obtain _K_
4. Decrypt _C_ using _K_ to obtain the plaintext

The parenthetical in the title is because it is at least potentially possible that the malware could be attacking Snappy implementations.

**Comment by mrcbax** [comment]
Yes I will look at it. If you can recommend any crates you feel are mature(ish) enough to have good docs let me know. The point of compressing is not just to "make safe" but also to reduce database size.
